### ISSUE TITLE
跑自己的数据做语义分割，跑不通

### ISSUE THREAD

**Issue by gang3217** [yanx27/Pointnet_Pointnet2_pytorch#162]
您好，我仿照S3DIS数据集做了自己的数据样本，但是用您的代码跑语义分割，到了 ”for i, (points, target) in tqdm(enumerate(trainDataLoader), total=len(trainDataLoader), smoothing=0.9):“就跑不动。跑S3DIS数据集没有问题。谢谢

**Comment by wyz5008** [comment]
你好，我也在尝试做自己的语义分割数据集，（已经采集做好了PCD格式的点云文件），请问你是如何打语义分割的标签，并且调整格式以及修改代码使其可以训练呢？

**Comment by yuhan1010** [comment]
hello，你有没有遇到 “ran out of input”这个error嘛。我用S3DIS跑就会出现这个问题

**Comment by hongzhiqiang67** [comment]
me too ,I have the same problem,Can you add me to discuss it,my qq is 3454136228

**Comment by hongzhiqiang67** [comment]
@gang3217 @wyz5008 @yushiangw

**Comment by zero406-CC** [comment]
你们应该看看这个S3DISDataLoader.py的 def __getitem__(self, idx):   if point_idxs.size > 1024:
                break   这个不同数据这个1024值不一样，自己的数据可能量小，这个随机取值的数量一直达不到1024 就会死循环。具体含义没细纠，但是这里我数据量小改了120（我的数据量差不多是这个，你们可以打印point_idxs.size评估看看）就跳出了，我刚学，其它不必问我，祝君好运。

**Comment by KungFuPandaMaster** [comment]
I have the same problem.May be my dataset too little?Which code can i change?
![image](https://user-images.githubusercontent.com/44471471/167100739-de400739-e803-44b8-9ab6-a482ed7d434f.png)

**Comment by zero406-CC** [comment]
your problem seems to be multithread problem on windows， trainDataLoader = torch.utils.data.DataLoader(TRAIN_DATASET, batch_size=BATCH_SIZE, shuffle=True, num_workers=0,
                                                  pin_memory=True, drop_last=True,
                                                  worker_init_fn=lambda x: np.random.seed(x + int(time.time())))
num_workers=0 try this，，你和他们的问题不一样，看着是windows下面的多线程问题

**Comment by mmiku1** [comment]
![J6I4)6~Z~ 13$D@9JZ~W3GN](https://user-images.githubusercontent.com/94426621/174422345-8307d578-bc91-4c01-ae1b-8cf33499d4cf.png)

**Comment by mmiku1** [comment]
请问我制作自己的数据集，遇到这个问题该怎么解决，我只放了少量数据进行尝试
![J6I4)6~Z~ 13$D@9JZ~W3GN](https://user-images.githubusercontent.com/94426621/174422351-3651a623-183a-48b0-aef8-5abb216d7266.png)

**Comment by wyz5008** [comment]
检查一下数据格式？看看npy文件是不是七维矩阵



------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年6月18日(星期六) 中午12:15
收件人: "yanx27/Pointnet_Pointnet2_pytorch";
抄送: "Mention";
主题: Re: [yanx27/Pointnet_Pointnet2_pytorch] 跑自己的数据做语义分割，跑不通 (Issue #162)





 
请问我制作自己的数据集，遇到这个问题该怎么解决，我只放了少量数据进行尝试
 
 
&mdash;

**Comment by mmiku1** [comment]
> 查看数据格式？查看文件不是七维矩阵
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: ***@***.***&gt;; 发送时间: 2022年6月18日(星期六) 中午12:15 收件人: "yanx27/Pointnet_Pointnet2_pytorch"; 抄送: "Mention"; 主题: Re: [yanx27/Pointnet_Pointnet2_pytorch] 跑自己的数据做语义分割，跑不通 (Issue #162) 请问我制作自己的数据集，遇到这个问题该怎么解决，我只放了少量数据进行尝试 &mdash; Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you were mentioned.Message ID: ***@***.***&gt;

数据格式没有问题，就是加载训练数据的时候会报错，加载测试数据的时候却没有错

**Comment by JayLuoShui** [comment]
> > 查看数据格式？查看文件不是七维矩阵
> > […](#)
> > ------------------ 原始邮件 ------------------ 发件人: _**@**_._**>; 发送时间: 2022年6月18日(星期六) 中午12:15 收件人: "yanx27/Pointnet_Pointnet2_pytorch"; 抄送: "Mention"; 主题: Re: [yanx27/Pointnet_Pointnet2_pytorch] 跑自己的数据做语义分割，跑不通 (Issue #162) 请问我制作自己的数据集，遇到这个问题该怎么解决，我只放了少量数据进行尝试 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you were mentioned.Message ID: _**@**_.**_>
> 
> 数据格式没有问题，就是加载训练数据的时候会报错，加载测试数据的时候却没有错

你好，请问如何标注自己的点云数据，有什么推荐的方法或教程，我在网上找了半天了，谢谢先

**Comment by sean880304** [comment]
> > > 查看数据格式？查看文件不是七维矩阵
> > > […](#)
> > > ------------------ 原始邮件 ------------------ 发件人: _**@**_._**>; 发送时间: 2022年6月18日(星期六) 中午12:15 收件人: "yanx27/Pointnet_Pointnet2_pytorch"; 抄送: "Mention"; 主题: Re: [yanx27/Pointnet_Pointnet2_pytorch] 跑自己的数据做语义分割，跑不通 (Issue #162) 请问我制作自己的数据集，遇到这个问题该怎么解决，我只放了少量数据进行尝试 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you were mentioned.Message ID: _**@**_.**_>
> > 
> > 
> > 数据格式没有问题，就是加载训练数据的时候会报错，加载测试数据的时候却没有错
> 
> 你好，请问如何标注自己的点云数据，有什么推荐的方法或教程，我在网上找了半天了，谢谢先

Use CloudCompare software.
1. Load scene point cloud into CloudCompare.
2. Use segment tool in the software to get object partial point cloud.
3. Export object point cloud with xyzrgb.
4. Run collect_indoor3d_data.py to label all points in the scene point cloud. (you will get a new point cloud data with x,y,z,r,g,b,l. l=label)

here is reference video of CloudCompare
https://youtu.be/-McuNKI-HHs

**Comment by XiaXinjie** [comment]
> > > 查看数据格式？查看文件不是七维矩阵
> > > […](#)
> > > ------------------ 原始邮件 ------------------ 发件人: _**@**_._**>; 发送时间: 2022年6月18日(星期六) 中午12:15 收件人: "yanx27/Pointnet_Pointnet2_pytorch"; 抄送: "Mention"; 主题: Re: [yanx27/Pointnet_Pointnet2_pytorch] 跑自己的数据做语义分割，跑不通 (Issue #162) 请问我制作自己的数据集，遇到这个问题该怎么解决，我只放了少量数据进行尝试 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you were mentioned.Message ID: _**@**_.**_>
> > 
> > 
> > 数据格式没有问题，就是加载训练数据的时候会报错，加载测试数据的时候却没有错
> 
> 你好，请问如何标注自己的点云数据，有什么推荐的方法或教程，我在网上找了半天了，谢谢先

您好，我也有关于制作数据集的问题，想要请教您，方便留个联系方式吗？我加您好友。

**Comment by wangzhen5201314** [comment]
> 
你好我也想联系一下制作数据集的问题可以加下好友吗 微信18018594107

**Comment by wangzhen5201314** [comment]
> > > > 查看数据格式？查看文件不是七维矩阵 [...](#) ------------------ 原始邮件 ------------------ 发件人： _**@**_._**>; 发送时间： 2022年6月18日（星期六） 中午12：15 收件人： “yanx27/Pointnet_Pointnet2_pytorch”; 抄送： “提及”; 主题： Re： [yanx27/Pointnet_Pointnet2_pytorch] 跑自己的数据做语义分割，跑不通 （Issue #162） 请问我制作自己的数据集，遇到这个问题该怎么解决，我只放了少量数据进行尝试 — 直接回复此电子邮件，在 GitHub 上查看或取消订阅。您收到此消息是因为您被提及。消息 ID：**_@**.**__>
> > > 
> > > 
> > > 数据格式没有问题，就是加载训练数据的时候会报错，加载测试数据的时候却没有错
> > 
> > 
> > 你好，请问如何标注自己的点云数据，有什么推荐的方法或教程，我在网上找了半天了，谢谢先
> 
> 您好，我也有关于制作数据集的问题，想要请教您，方便留个联系方式吗？我加您好友。
你好可以加下好友吗  我也有类似问题 微信是18018594107 @yuhan1010 @KungFuPandaMaster @zero406-CC @JayLuoShui @
